cli/src/torii/utils.rs

```diff
@@ -90,27 +91,4 @@ impl<O: Reply, E: Reply> Reply for WarpResult<O, E> {
     }
 }
 
-macro_rules! impl_custom_and_then {
-    ( $name:ident ( $($arg_name:ident : $arg_gen:ident),* $(,)? ) ) => {
-        /// Maps filter to handler with `n` arguments (`n` is suffix of function)
-        pub fn $name<O, E, F, Fut, Fil, $($arg_gen,)*>(f: F, router: Fil)
-            -> impl Filter<Extract = (WarpResult<O, E>,), Error = Rejection> + Clone
-        where
-            Fil: Filter<Extract = ($($arg_gen,)*), Error = Rejection> + Clone,
-            F: Fn($($arg_gen,)*) -> Fut + Copy + Send + Sync + 'static,
-            Fut: std::future::Future<Output = Result<O, E>> + Send,
-            $($arg_gen: Send,)*
-        {
-            router.and_then(move |$($arg_name,)*|
-                async move {
-                    Ok::<_, Infallible>(WarpResult(f($($arg_name,)*).await))
-                }
-            )
-        }
-    }
-}
-
-// impl_custom_and_then!(endpoint1(a: A));
-impl_custom_and_then!(endpoint2(a: A, b: B));
-impl_custom_and_then!(endpoint3(a: A, b: B, c: C));
-impl_custom_and_then!(endpoint4(a: A, b: B, c: C, d: D));
+generate_endpoints_up_to!(4);
```

Comment:
I'd prefer a fn-like macro that works a little more like 
```
generate_endpoints!(2, 3, 4);
```

The reasoning is as follows: the function that you generate is public, so unless you have LTO enabled, the `endpoint1` function will be included in the binary and bloat it. We only want to have the `endpoint` functions which are actually used.

Reply:
then I would suggest something along the lines of:
```suggestion
generate_endpoint!(endpoint_name1, 3);
generate_endpoint!(endpoint_name2, 4);
```

where you define the name of the endpoint, and the number of arguments it receives

Reply:
Do we really need the option of defining custom names for endpoints? This doesn't seem to bring any semantic value, as anyway they're wrapping handlers with more descriptive names, and there's a bit more boilerplate involved if we generate just one endpoint per macro invocation.

Reply:
I don't think we need the custom names just yet. But it is something that we should keep in mind. If we did do that, we could use a syntax that's a little more unambiguous: e.g. `name: 1, name_2: 2`...